schema/crdb/46.0.0/up2.sql

```diff
@@ -0,0 +1,17 @@
+-- Ensure there are no `sled_resource` rows with a `kind` other than 'instance'
+
+-- This is a full table scan, but the sled_resource table does not track
+-- historical, deleted resources, so is at most the size of the number of
+-- currently-running instances (which should be zero during a schema update).
+SET
+  LOCAL disallow_full_table_scans = OFF;
```

Comment:
Should this be the first part of the schema change, before dropping the service table?

Reply:
Yes, yes it should. Alternatively, we could drop it entirely:

* I checked dogfood and Augustus checked colo to confirm all `sled_resource`s are `instance`.
* Removing the enum variants on the Rust side had no followon effects (i.e., we were never creating them in real code or tests)

so it's probably overly cautious to include?

Reply:
Swapped up1 and up2 in dc18f5758